accelerator/xpu_accelerator.py

```diff
@@ -253,33 +257,29 @@ def on_accelerator(self, tensor):
         else:
             return False
 
+    def _lazy_init_class_dict(self):
+        if self.class_dict:
+            return
+
+        op_builder_module = importlib.import_module(self.op_builder_dir())
+
+        # get op builder class from op_builder/xpu/__init__.py
+        self.class_dict = {}
+        for class_name, class_obj in inspect.getmembers(op_builder_module, inspect.isclass):
+            self.class_dict[class_name] = class_obj
+
     # create an instance of op builder and return, name specified by class_name
-    def create_op_builder(self, op_name):
-        builder_class = self.get_op_builder(op_name)
-        if builder_class != None:
-            return builder_class()
-        return None
+    def create_op_builder(self, class_name):
+        builder_class = self.get_op_builder(class_name)
+        return None if builder_class is None else builder_class()
```

Comment:
builder_class could return NotImplementedBuilder or None. It's error prone to  check between the two.

Reply:
@YizhouZ thoguhts on this?

Reply:
> @YizhouZ thoguhts on this?

Previously we didn't add NoneImpl builder. Will remove this line to only return NoneImpl builder.